### DISCUSSION TITLE
Growatt & Venus OS

### DISCUSSION THREAD

**Discussion by MichaelRayHam** [Louisvdw/dbus-serialbattery#258] [Q&A]
Would like to see if you would be interested in writing code to allow a Growatt Inverter to communicate with Venus OS.  I would be willing sponsor this project.  Please let me know if this is something you would be interested in.  Thank you

**Comment by ganey** [comment]
If the inverter has rs485/canbus, you can connect it to venus then run some scripts to publish the inverter values on DBUS.

These are the available options for inverters on dbus: https://github.com/victronenergy/venus/wiki/dbus#inverter

I noticed this project (https://github.com/nygma2004/growatt2mqtt) that makes the inverter details available on MQTT for growwatt, it's probably not yet compatible with venus MQTT, but might be worth asking over there too?

**Comment by Louisvdw** [comment]
I won't have the capacity to tackle something like that at this stage. But it does sound interesting

**Comment by MichaelRayHam** [comment]
Thank you so much for the feedback.

Here’s some information on my communications configuration:  My battery is managed by a Batrium of which is monitoring 8 separate batteries via the Batrium system.  The Batrium is communication with my Victron Cerbo GX and is throttling and controlling my charge controllers (Victron) and it is working perfectly, no issues.  I am able to see the Growatt inverter output using Energy Meters which are connected to the Cerbo GX, no issues.  I am also able to see my AC input into the Growatt inverters should the situation call for grid input.  I am again using Energy Meters to communication this information to the Cerbo GX, no issues.  The information I need to send back to my Growatt is the State of Charge of the battery.  The reason I need the Growatts to see the SOC of the battery is to switch from battery to grid should I drop below a State of Charge percentage that I determine.  I would also like the Cerbo GX to be able to show the status of the inverters such as On/Off position, errors, ect.

I will look into the two options below and maybe someone can help me with this.

Thanks to you guys for all you contribute to the community.

Michael Ham
Project Ham

**Comment by ch-reck** [comment]
I'm running for many weeks the [grott](https://github.com/johanmeijer/grott) proxy to side feed a local MQTT - this is a great solution that does not need any firmware change on the shine stick, only changing the growatt-server IP to point to the proxy.

From there I've adapted the [dbus-venus-mqtt-pvinverter.py](https://github.com/tbinias/dbus-venus-mqqt-pvinverter/blob/main/dbus-venus-mqtt-pvinverter.py) to take the [grott ](https://github.com/johanmeijer/grott) MQTT feed from my Growatt MIN2500TL 1 phase inverter and pass the values to the victron DBus. This is working like a charm!

My adapted code is here:
https://github.com/ch-reck/dbus-mqtt-pv-grott
